core/state/interface.go

```diff
@@ -0,0 +1,104 @@
+package state
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/stateless"
+	"github.com/ethereum/go-ethereum/core/tracing"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/types/bal"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/trie/utils"
+	"github.com/holiman/uint256"
+)
+
+// BlockProcessingDB encapsulates all functionality necessary to fully execute
+// a block.  It is an amalgamation of state access/modification functionality
+// utilized by EVM execution and other functionality needed in block execution
+// (e.g. state root calculation).
+type BlockProcessingDB interface {
```

Comment:
Most of the methods here are the same as `vm.StateDB`, can we somehow combine these interfaces? So that we don't need to change 2 places if anything changes.

Reply:
Moving it into `core` also causes an import cycle. Ugh